pyquil/api/_engagement_manager.py

```diff
@@ -44,28 +45,37 @@ def __init__(self, *, client_configuration: QCSClientConfiguration) -> None:
         :param client_configuration: Client configuration, used for refreshing engagements.
         """
         self._client_configuration = client_configuration
-        self._cached_engagements: Dict[str, EngagementWithCredentials] = {}
+        self._cached_engagements: Dict[Tuple[str, Optional[str]], EngagementWithCredentials] = {}
         self._lock = threading.Lock()
 
-    def get_engagement(self, *, quantum_processor_id: str, request_timeout: float = 10.0) -> EngagementWithCredentials:
+    def get_engagement(
+        self, *, quantum_processor_id: str, request_timeout: float = 10.0, endpoint_id: Optional[str] = None
+    ) -> EngagementWithCredentials:
         """
-        Gets an engagement for the given quantum processor. If an engagement was already fetched previously and
-        remains valid, it will be returned instead of creating a new engagement.
+        Gets an engagement for the given quantum processor endpoint.
+        If an engagement was already fetched previously and remains valid, it will be returned instead
```

Comment:
```suggestion
        
        If an engagement was already fetched previously and remains valid, it will be returned instead
```
Sphinx will want you to have an extra line between the summary and the body. Sometimes it renders things weird without the blank.